### DISCUSSION TITLE
How to add to the turbulence equation a variable

### DISCUSSION THREAD

**Discussion by TTtianhong** [su2code/SU2#2403] [Q&A]
How to add to the turbulence equation a variable that I have provided externally that has a spatial distribution of coordinates and values （Constant values in each point，different in space but the same in time）:

First I tried adding a new column of variables （bbeta）to restart_flow.csv.
Then:    I used this to load variables: 
    if (incompressible && ((!energy) && (!weakly_coupled_heat))) skipVars--;

    /*--- Load data from the restart into correct containers. ---*/

    unsigned long counter = 0;
    for (auto iPoint_Global = 0ul; iPoint_Global < geometry[MESH_0]->GetGlobal_nPointDomain(); iPoint_Global++) {
      /*--- Retrieve local index. If this node from the restart file lives
       on the current processor, we will load and instantiate the vars. ---*/

      const auto iPoint_Local = geometry[MESH_0]->GetGlobal_to_Local_Point(iPoint_Global);

      if (iPoint_Local > -1) {
        /*--- We need to store this point's data, so jump to the correct
         offset in the buffer of data from the restart file and load it. ---*/

        const auto index = counter * Restart_Vars[1] + skipVars;
        for (auto iVar = 0u; iVar < nVar; iVar++) nodes->SetSolution(iPoint_Local, iVar, Restart_Data[index + iVar]);

        /*--- Increment the overall counter for how many points have been loaded. ---*/
        counter++;
      }

I want to use the variable in SA model：

   /*! \brief Baseline (Original SA model). */
struct Bsl {
  static void get(const su2double& nue, const CSAVariables& var, su2double& production, su2double& destruction,
                  su2double& cross_production, su2double& jacobian) {
    ComputeProduction(nue, var, production, jacobian);
    ComputeDestruction(nue, var, destruction, jacobian);
    ComputeCrossProduction(nue, var, cross_production, jacobian);
  }

  static void ComputeProduction(const su2double& nue, const CSAVariables& var, su2double& production,
                                su2double& jacobian) {
    const su2double factor = var.intermittency * var.cb1;
    production = factor * (1.0 - var.ft2) * var.Shat * nue;
    jacobian += factor * (-var.Shat * nue * var.d_ft2 + (1.0 - var.ft2) * (nue * var.d_Shat + var.Shat));
  }
...

How can I add the new variable such as "bbeta" to the calculation of "ComputeProduction" to make new factor = bbeta *factor;

**Comment by pcarruscag** [comment]
In CTurbSASolver::Source_Residual you need to pass your variable to the numerics class, see how any of the other `Set` methods are implemented.
Then in CSourceBase_TurbSA you can put this new variable in CSAVariables to be able to use it in the different terms of the model.

**Reply by Nuguri22** [reply]
Dear pcarruscag

Your advice is very helpful. Thank you very much for you detailed reply.
